app/backend/lib/sharepoint.ts

```diff
@@ -0,0 +1,65 @@
+import { Router } from 'express';
+import * as XLSX from 'xlsx';
+import * as spauth from 'node-sp-auth';
+import config from '../../config';
+import getAuthRole from '../../utils/getAuthRole';
+
+const SP_SITE = config.get('SP_SITE');
+const SP_DOC_LIBRARY = config.get('SP_DOC_LIBRARY');
+const SP_FILE_NAME = config.get('SP_MS_FILE_NAME');
+const SP_SA_USER = config.get('SP_SA_USER');
+const SP_SA_PASSWORD = config.get('SP_SA_PASSWORD');
+
+const sharepoint = Router();
+
+// eslint-disable-next-line consistent-return
+sharepoint.get('/api/sharepoint/masterSpreadsheet', (req, res) => {
+  const authRole = getAuthRole(req);
+  const isRoleAuthorized = authRole?.pgRole === 'ccbc_admin';
```

Comment:
Not needed for now, but do we want this to only be accessible to admins?

Reply:
Agreed, I didn't want this to be open without any kind of role checks. 

I was thinking we can get the service account to get a token with the proper role, this can be admin, analyst or a custom one only for the service account.